service/StudentService.java

```diff
@@ -33,16 +34,21 @@ public void deleteStudent(String studentId){
         studentDao.delete(studentId);
     }
 
-    public List<Student> sortStudentsById(){
-        return studentDao.sortStudentsById();
-    }
-
-    public List<Student> sortStudentsByFirstName(){
-        return studentDao.sortStudentsByFirstName();
-    }
-
-    public List<Student> sortStudentsByLastName(){
-        return studentDao.sortStudentsByLastName();
+    public List<Student> sortStudents(int choice){
+        switch (choice){
+            case 1 -> {
+                return studentDao.sortStudents(Comparator.comparing(Student::getStudentId));
+            }
+            case 2 -> {
+                return studentDao.sortStudents(Comparator.comparing(Student::getFirstName));
+            }
+            case 3 -> {
+                return studentDao.sortStudents(Comparator.comparing(Student::getLastName));
+            }
+            default -> {
+                return null;
```

Comment:
Avoid returning null. If null really has a meaning, then return an `Optional<OF_SOMETHING>`. In this case you could make use of Exceptions :) so you can throw an exception and handle it in the presentation layer.

Reply:
can i remove the default block totally?

Reply:
You can leave the default block, but instead of `return null;` you can write `throw new ...Exception()` and handle the exception accordingly in the presentation layer.